tests/e2e/common.go

```diff
@@ -0,0 +1,733 @@
+		deployGrafana:           false,
+		deployPrometheus:        false,
+		deployJaeger:            false,
+
+		vaultHost:     "vault." + td.osmMeshName + ".svc.cluster.local",
+		vaultProtocol: "http",
+		vaultRole:     "openservicemesh",
+		vaultToken:    "token",
+	}
+}
+
+// InstallOSM installs OSM. Right now relies on externally calling the binary and a subset of possible opts
+// TODO: refactor install to be able to call it directly here vs. exec-ing CLI.
+func (td *OsmTestData) InstallOSM(instOpts InstallOSMOpts) error {
+	if td.kindCluster {
+		td.T.Log("Getting image data")
+		imageNames := []string{
+			"osm-controller",
+			"init",
+		}
+		docker, err := client.NewClientWithOpts(client.WithAPIVersionNegotiation())
+		if err != nil {
+			return errors.Wrap(err, "failed to create docker client")
+		}
+		var imageIDs []string
+		for _, name := range imageNames {
+			imageName := fmt.Sprintf("%s/%s:%s", td.ctrRegistryServer, name, td.osmImageTag)
+			imageIDs = append(imageIDs, imageName)
+		}
+		imageData, err := docker.ImageSave(context.TODO(), imageIDs)
+		if err != nil {
+			return errors.Wrap(err, "failed to get image data")
+		}
+		defer imageData.Close()
+		nodes, err := td.clusterProvider.ListNodes(td.clusterName)
+		if err != nil {
+			return errors.Wrap(err, "failed to list kind nodes")
+		}
+		for _, n := range nodes {
+			td.T.Log("Loading images onto node", n)
+			if err := nodeutils.LoadImageArchive(n, imageData); err != nil {
+				return errors.Wrap(err, "failed to load images")
+			}
+		}
+	}
+
+	if err := td.CreateNs(instOpts.controlPlaneNS, nil); err != nil {
+		return errors.Wrap(err, "failed to create namespace "+instOpts.controlPlaneNS)
+	}
+
+	if instOpts.certManager == "vault" {
+		if err := td.installVault(instOpts); err != nil {
+			return err
+		}
+	}
+
+	td.T.Log("Installing OSM")
+	var args []string
+
+	// Add OSM namespace to cleanup namespaces, in case the test can't init
+	td.cleanupNamespaces[instOpts.controlPlaneNS] = true
+
+	args = append(args, "install",
+		"--container-registry="+instOpts.containerRegistryLoc,
+		"--osm-image-tag="+instOpts.osmImagetag,
+		"--namespace="+instOpts.controlPlaneNS,
+		"--certificate-manager="+instOpts.certManager,
+		"--enable-debug-server",
+	)
+
+	switch instOpts.certManager {
+	case "vault":
+		args = append(args,
+			"--vault-host="+instOpts.vaultHost,
+			"--vault-token="+instOpts.vaultToken,
+			"--vault-protocol="+instOpts.vaultProtocol,
+			"--vault-role="+instOpts.vaultRole,
+		)
+	}
+
+	if len(instOpts.containerRegistrySecret) != 0 {
+		args = append(args, "--container-registry-secret="+registrySecretName)
+	}
+
+	args = append(args, fmt.Sprintf("--enable-prometheus=%v", instOpts.deployPrometheus))
+	args = append(args, fmt.Sprintf("--enable-grafana=%v", instOpts.deployGrafana))
+	args = append(args, fmt.Sprintf("--deploy-jaeger=%v", instOpts.deployJaeger))
+
+	stdout, stderr, err := td.RunLocal(filepath.FromSlash("../../bin/osm"), args)
```

Comment:
Is there a path to also test `helm install`

Reply:
Not yet, but we have scenario documented in our running list: https://github.com/openservicemesh/osm/issues/1714#issuecomment-703843060

Reply:
As discussed offline, `helm install` should be part of the test suite regardless of the scenarios we are testing. Added https://github.com/openservicemesh/osm/issues/1814 to track.